# IS362 - Week 5 Assignment

Your task in this week’s assignment is to answer three questions:
1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

The data for this assignment has been provided in another repository, so we can start by importing the pandas module:

In [1]:
import pandas as pd

The next think we're going to do is read the data that has been provided regarding the US airports.

In [2]:
airports = pd.read_csv("https://raw.githubusercontent.com/maddogdatascience/DATA607/master/airports.csv")

The first question I need an answer to is what is the northernmost airport in the United States? In order to do this, I want to sort the airports in order of latitude, because this will tell me which airport is the furthest north. Because we're in the Northern Hemisphere, the airport with the largest latitude will be the furthest north. Let's look at the top 5:

In [3]:
airports.sort_values(by=["lat"], ascending=False).head()

faa                                        name        lat         lon  \
393  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
217  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
98   AIN                          Wainwright Airport  70.638056 -159.994722   
674  K03                               Wainwright As  70.613378 -159.860350   
140  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst  
393  149  -5   A  
217   44  -9   A  
98    41  -9   A  
674   35  -9   A  
140   96  -9   A

However, when I confirm with Google, Dillant Hopkins Airport is not the northermost airport. Dillant Hopkins Airport is in New Hampshire, and its coordinates are 42.9064° N, 72.2726° W. This is the reverse of what is reflected in the CSV. When this error is accounted for, the Wiley Post Will Rogers Memorial Airport located in Alaska, is the northernmost airport. This was confirmed with Google. 

The next question is what is the easternmost airport in the United States? For this, we want to sort the airports in order of longitude rather than latitude, as this will tell us which is the furthest east. We'll use .head() to look at the top 5 again in order to save space:

In [16]:
airports.sort_values(by=["lon"], ascending=False).head()

faa                           name        lat         lon   alt  tz dst
1237  SYA                   Eareckson As  52.712275  174.113620    98 -10   A
897   MYF               Montgomery Field  32.475900  117.759000    17  -8   A
372   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478  -7   A
393   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A
419   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A

Once again, a quick Google search tells us that the data is once again incorrect. Eareckson Air Station is located in Alaska - definitely not the easternmost state in the US. However, if we look at the longitude values, we see that they are missing something: a minus sign. Because the longitude is in the western hemisphere, it is represented with a minus sign, which is missing from the first four airports in the list. Once we account for this error, Eastport Municipal Airport is the easternmost airport, having the lowest longitude. The name certainly sounds more promising, and Google will again confirm that this is correct. 

The last question is which airport had the windiest weather on February 12th, 2013. For this, I need to read a different CSV containing data on the weather.

In [12]:
weather = pd.read_csv("https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv")

I'm only concerned about the weather on one day, February 12th of the year 2013, so I'm only going to display values for that day, and then sort them by who has the most significant wind gusts:

In [13]:
weather[(weather["month"]==2) & (weather["day"]==12) & (weather["year"]==2013)].sort_values(by=["wind_gust"], ascending=False).head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

It would appear that Newark Airport wins by a landslide, but if we look at the data, the values don't make sense. At 1048 and 1206, the values are astronimically larger than the next closest airport, which is only about an hour away. Furthermore, the data presented was tracked multiple times a day, meaning we have readings for the morning, and the evening. While for the next closest airport, the values are reasonably close, for Newark, they are incredibly far apart. It appears that there is another error in the data, and that 

# Final Answers:

1. What is the northernmost airport in the United States? - Wiley Post Will Rogers Memorial Airport
2. What is the easternmost airport in the United States? - Eastport Municipal Airport
3. On February 12th, 2013, which New York area airport had the windiest weather? - LaGuardia Airport